lib/rex/socket/comm/local.rb

```diff
@@ -440,76 +424,49 @@ def self.proxy(sock, type, host, port)
         raise Rex::ConnectionProxyError.new(host, port, type, "Failed to receive a response from the proxy"), caller
       end
 
-      resp = Rex::Proto::Http::Response.new
-      resp.update_cmd_parts(ret.split(/\r?\n/)[0])
+      if (match = ret.match(/HTTP\/.+?\s+(\d+)\s?.+?\r?\n?$/))
+        status_code  = match[1].to_i
+      end
```

Comment:
This fixed a bug where an exception would be raised if Rex::Socket was used out side the context of Metasploit because `Rex::Proto::Http::Response` is unavailable. The other proxies just do the bare minimum parsing necessary to ensure the connection was established, so that patter was copied here as well. Removing this dependency also means the HTTP proxy can be tested in the specs.